google-cloud-storage/src/main/java/com/google/cloud/storage/GapicBidiUnbufferedWritableByteChannel.java

```diff
@@ -345,10 +346,17 @@ public void onNext(BidiWriteObjectResponse value) {
     public void onError(Throwable t) {
       if (t instanceof OutOfRangeException) {
         OutOfRangeException oore = (OutOfRangeException) t;
-        clientDetectedError(
-            ResumableSessionFailureScenario.SCENARIO_5.toStorageException(
-                ImmutableList.of(lastWrittenRequest), null, context, oore));
-      } else if (t instanceof ApiException) {
+        ErrorDetails ed = oore.getErrorDetails();
+        if (!(ed != null
+            && ed.getErrorInfo() != null
+            && ed.getErrorInfo().getReason().equals("GRPC_MISMATCHED_UPLOAD_SIZE"))) {
+          clientDetectedError(
+              ResumableSessionFailureScenario.SCENARIO_5.toStorageException(
```

Comment:
can we get some comments around what Scenario 5 is to make it easier to read?

Reply:
oh i see this was there before as well. my apologies. that can be a followup.

Reply:
ResumableSessionFailureScenario is an enum that is shared by several of our upload implementations and defines the types of failure scenarios that can happen when processing an upload request.

SCENARIO_5 is client side data loss detected https://github.com/googleapis/java-storage/blob/380057bd5c6f74158e55854dc159382703ea328f/google-cloud-storage/src/main/java/com/google/cloud/storage/ResumableSessionFailureScenario.java#L72-L75